sourcecode/apis/common/app/Http/Controllers/AuthMigrationController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\AuthMigration;
+use App\Http\Requests\CreateAuthMigrationRequest;
+use Cerpus\LaravelRabbitMQPubSub\Facades\RabbitMQPubSub;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Support\Facades\Cache;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\Exception\PreconditionFailedHttpException;
+
+final class AuthMigrationController extends Controller
+{
+    public function get(string $id): JsonResponse
+    {
+        $this->authorize('admin');
+
+        /** @var AuthMigration $authMigrationData */
+        $authMigrationData = Cache::get('auth-migration-' . $id);
+
+        if ($authMigrationData == null) {
+            throw new NotFoundHttpException("Auth migration was not found");
+        }
+
+        if (count($authMigrationData->tables) == 7) {
```

Comment:
What is the significance of the number 7?

I suggest this check be encapsulated as part of the data class, i.e. something like `$authMigrationData->isReady()`. If the logic ever needs to be updated in the future, this could be done in a single place.